mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/PojoWorker.java

```diff
@@ -15,16 +15,64 @@
  */
 public interface PojoWorker {
 
+	/**
+	 * Add an entity to the index, assuming that the entity is absent from the index.
+	 * <p>
+	 * Shorthand for {@code add(null, entity)}; see {@link #add(Object, Object)}.
+	 *
+	 * @param entity The entity to add to the index.
+	 */
 	void add(Object entity);
 
+	/**
+	 * Add an entity to the index, assuming that the entity is absent from the index.
+	 * <p>
+	 * <strong>Note:</strong> depending on the backend, this may lead to a corrupted index
```

Comment:
I would have expected potential errors but corrupted indexes?

Reply:
Ok, maybe not corrupted index, but at least wrong data in the index. Like, two documents with the same ID (in the Hibernate Search sense). The Lucene document ID would be different, though, so you're right, the index wouldn't really be "corrupted".

Changed to:

```
	 * <strong>Note:</strong> depending on the backend, this may lead to errors or duplicate entries in the index
	 * if the entity was actually already present in the index before this call.
```